### PR TITLE
E2e multiple erc20 watch asset

### PR DESCRIPTION
**Pull request by tmashuang**
## **Description**
_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

Closes https://github.com/MetaMask/MetaMask-planning/issues/1407

This test involves handing multiple erc20 tokens on the suggested watch asset notification. Currently this tests current behavior of prod, where only the last of multiple tokens in the watchAsset confirmation will be added to the token list. With the future bump of the `@metamask/assets-controller` this test's assertion will have to change when multiple tokens are properly added.

## **Manual testing steps**

_1. Pull down this branch
_2. `yarn && yarn build:test && yarn test:e2e:single test/e2e/tests/add-multiple-tokens.spec.js --browser=${BROWSER}`
_3. Should not fail. But when bumping `@metamask/assets-controller` should fail.

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/add-multiple-tokens.spec.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        assert.equal(multipleSuggestedtokens.length, 3);
+        await driver.findClickableElement({ text: 'Add token', tag: 'button' });
+        await driver.clickElement({ text: 'Add token', tag: 'button' });
+
+        // Switch to fullscreen extension
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        // Check only 1 token is added bug.
```
Comment (adonesky1) [inline test/e2e/tests/add-multiple-tokens.spec.js:101]: We should probably wait til this bug is fixed and merge the test with the expected behavior asserted?

**Comment by adonesky1** [comment]
Nice so the test is failing now with the upgrade that includes the fix! So just need to update the tests to reflect the fixed behavior!

**Review by adonesky1 — DISMISSED** [review]
LGTM!

File: test/e2e/tests/add-multiple-tokens.spec.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+        // Wait for token 1 address to populate in dapp
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await driver.wait(async () => {
+          const tokenAddressesElement = await driver.findElement(
+            '#tokenAddresses',
+          );
+          const tokenAddresses = await tokenAddressesElement.getText();
+          return tokenAddresses !== '';
+        }, 20000);
```
Comment (jiexi) [inline test/e2e/tests/add-multiple-tokens.spec.js:45]: 20s seems long if we're only waiting 10s in the condition below this one

Comment (tmashuang) [reply test/e2e/tests/add-multiple-tokens.spec.js]: Resolved in [604f46d](https://github.com/MetaMask/metamask-extension/pull/21381/commits/604f46dcd9cd0bf71892ba470a940515ac3191bf)

**Review by adonesky1 — APPROVED** [review]

**Review by jiexi — APPROVED** [review]

**Pull request merged**